code/workspaces/client-api/src/schema/resolvers.js

```diff
@@ -82,6 +83,7 @@ const resolvers = {
     setInstanceAdmin: (obj, { userId, instanceAdmin }, { user, token }) => instanceAdminWrapper(user, () => userService.setInstanceAdmin(userId, instanceAdmin, token)),
     setDataManager: (obj, { userId, dataManager }, { user, token }) => instanceAdminWrapper(user, () => userService.setDataManager(userId, dataManager, token)),
     setCatalogueRole: (obj, { userId, catalogueRole }, { user, token }) => instanceAdminWrapper(user, () => userService.setCatalogueRole(userId, catalogueRole, token)),
+    createCluster: (obj, { cluster }, { user, token }) => projectPermissionWrapper({ projectKey: cluster.projectKey }, STACKS_CREATE, user, () => clustersService.createCluster(cluster, token)),
```

Comment:
I've started leaving out the permission wrappers in the client api as it is handled by the infrastructure api and means there is only a single place to define the permissions required to perform an action in the back-end. I don't know whether it is worth doing the same here?

Reply:
Agreed